src/components/overview/StyleSelector.jsx

```diff
@@ -2,41 +2,47 @@ import React, {useState} from 'react';
 import styles from './overview.module.css';
 import json from './styleSample.json';
 import {FaCheck} from 'react-icons/fa';
+import {StyleIndexContext} from './overview.jsx'
+
 
 const StyleSelector = (props) => {
-  var data = json;
-  var styleOptions = data.results //all results of styles form ID, sample data
-  const [selectedStyle, setStyle] = useState(styleOptions[0]);
 
+  console.log(props.styles)
+  const [selectedStyle, setStyle] = useState(props.styles[0]);
 
   const handleClick = (event, option) => {
-
     setStyle(option);
-    //TODO
-    // Change state of Image gallery if this option changes
   }
 
-  //For every style option, display the first photo thumbnail of the option
-  return (
-    <>
-
-      <p className={styles.styleText}>STYLE > <span>{selectedStyle.name}</span></p>
 
-      <div className={styles.styleSelection}>
-
-      {styleOptions.map( (option, key) =>
-        <span className={styles.styleItem} key={key}>
-          <img
-            className={styles.styleThumbnail}
-            src={option.photos[0].thumbnail_url}
-            onClick={(e)=>{handleClick(e, option)}}
-          />
-          {option === selectedStyle && <FaCheck id={styles.check} color="#00acee"/>}
-        </span>
+  return (
+    <StyleIndexContext.Consumer>
```

Comment:
If this is working fine for you then it must be alright to do it this way. From what I understand if you have a class component then you have to use `Context.Consumer` this way but a more generally excepted way to do it (and arguably cleaner) is to use a useContext hook.

To do that you would import useContext in the same way you import useState and then at the top of the functional component you would use the hook in a very similar way to how you use useState....

However looking at your Overview it looks like you are using a different method than I used and if it is working that is great because it looks like a simpler method than what I used. If I don't comprehend what is going on by the time I finish this review I might have you walk me through what's going on!